bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs

```diff
@@ -98,8 +101,32 @@ public async Task<Provider> CompleteSetupAsync(Provider provider, Guid ownerUser
             throw new BadRequestException("Invalid owner.");
         }
 
-        provider.Status = ProviderStatusType.Created;
-        await _providerRepository.UpsertAsync(provider);
+        var consolidatedBillingEnabled = _featureService.IsEnabled(FeatureFlagKeys.EnableConsolidatedBilling);
+
+        if (!consolidatedBillingEnabled)
+        {
+            provider.Status = ProviderStatusType.Created;
+            await _providerRepository.UpsertAsync(provider);
+        }
+        else
+        {
+            if (taxInfo == null || string.IsNullOrEmpty(taxInfo.BillingAddressCountry) || string.IsNullOrEmpty(taxInfo.BillingAddressPostalCode))
+            {
+                throw new BadRequestException("Both address and postal code are required to set up your provider.");
+            }
+
+            var customer = await _providerBillingService.SetupCustomer(provider, taxInfo);
+
+            provider.GatewayCustomerId = customer.Id;
+
+            var subscription = await _providerBillingService.SetupSubscription(provider);
+
+            provider.GatewaySubscriptionId = subscription.Id;
+
+            provider.Status = ProviderStatusType.Billable;
+
+            await _providerRepository.UpsertAsync(provider);
+        }
```

Comment:
⛏️I feel like this block of code is formatted a bit weird. I'd like to suggest removing the variable assignment of the feature flag, and instead using checking for the flag directly in the conditional. I'd also like to suggest removing the white space between lines in the `else` block.
```suggestion
        if (!(_featureService.IsEnabled(FeatureFlagKeys.EnableConsolidatedBilling)))
        {
            provider.Status = ProviderStatusType.Created;
            await _providerRepository.UpsertAsync(provider);
        }
        else
        {
            if (taxInfo == null || string.IsNullOrEmpty(taxInfo.BillingAddressCountry) || string.IsNullOrEmpty(taxInfo.BillingAddressPostalCode))
            {
                throw new BadRequestException("Both address and postal code are required to set up your provider.");
            }
            var customer = await _providerBillingService.SetupCustomer(provider, taxInfo);
            provider.GatewayCustomerId = customer.Id;
            var subscription = await _providerBillingService.SetupSubscription(provider);
            provider.GatewaySubscriptionId = subscription.Id;
            provider.Status = ProviderStatusType.Billable;
            await _providerRepository.UpsertAsync(provider);
        }
```